gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,162 @@
+
+        Args:
+            folder: the base folder which contains image sequence
+            image_extension: file extension for the image files. Defaults to 'jpg'.
+            max_resolution: integer representing maximum length of image's short side, i.e.
+                the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
+                max_resolution would be 1080. If the image resolution max(height, width) is
+                greater than the max_resolution, it will be downsampled to match the max_resolution.
+            max_num_imgs: integer representing max number of images to process in the sequence,
+                0 or negative values means to process all the images
+        """
+        super().__init__(max_resolution=max_resolution)
+
+        # fetch all the file names in /images folder
+        search_path = os.path.join(folder, 'images', f"*.{image_extension}")
+
+        # @NOTE for now we only process images with valid EXIF data
+        self._image_paths, self._num_exif_imgs, self._num_all_imgs = \
+            self.get_images_with_valid_exif(search_path, max_num_imgs)
+        self._num_imgs = len(self._image_paths)
+
+        # self._image_paths.sort()
+        print(f'selected {self._num_imgs} out of {self._num_exif_imgs} images with valid exif')
+
+        if self._num_imgs == 0:
+            raise RuntimeError(
+                f"Loader could not find any images with the specified file extension in {search_path}")
+
+    def get_images_with_valid_exif(self, search_path: str, max_num_imgs: int) -> List[str]:
+        """Return a subset of images with valid exif.
+
+        Args:
+            search_path: image sequence search path
+            max_num_imgs: the maximum number of images to process
+
+        Returns:
+            list of selected image path.
+            the number of images with valid exif data.
+            the number of all the images.
+        """
+
+        all_image_paths = glob.glob(search_path)
+        num_all_imgs = len(all_image_paths)
+        exif_image_paths = []
+        for p in all_image_paths:
+            # drop images without valid exif data
+            if io_utils.load_image(p).get_intrinsics_from_exif():
+                exif_image_paths.append(p)
+
+        # randomly select a subset to process
+        num_exif_imgs = len(exif_image_paths)
+        max_num_imgs = max_num_imgs if (max_num_imgs > 0 and max_num_imgs < num_exif_imgs) else num_exif_imgs
+        random_indices = np.random.permutation(num_exif_imgs)
+        return np.array(exif_image_paths)[random_indices[:max_num_imgs]], num_exif_imgs, num_all_imgs
+
+    def image_filenames(self) -> List[str]:
+        """Return the file names corresponding to each image index."""
+        return [Path(fpath).name for fpath in self._image_paths]
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            the number of images.
```

Comment:
nit capitalization: "The number of images."

Reply:
Done